### DISCUSSION TITLE
Build Acala From Source

### DISCUSSION THREAD

**Discussion by andreclaro** [AcalaNetwork/Acala#2828] [0. General]
How to build acala and karura from source?

**Comment by andreclaro** [comment]
@zjb0807 I have used `cargo build --features with-{{chain}}-runtime` but there is no acala binary inside the target folder.

**Reply by andreclaro** [reply]
I also tried:

```
cargo build --release --locked --features with-acala-runtime --features try-runtime --bin acala
error: no bin target named `acala`.
```

**Reply by andreclaro** [reply]
```
 make build-release
make: *** No rule to make target 'build-release'.  Stop.
```

**Reply by zjb0807** [reply]
we use the https://github.com/AcalaNetwork/acala-node  to build the node now.

you can download the repo and refer to the Dockerfile
https://github.com/AcalaNetwork/acala-node/blob/master/Dockerfile#L7

**Reply by andreclaro** [reply]
But that repository is using a different release version: https://github.com/AcalaNetwork/acala-node/releases

**Reply by andreclaro** [reply]
Should we following https://github.com/AcalaNetwork/acala-node/releases instead for new node upgrades for acala and karura?

**Reply by zjb0807** [reply]
> Should we following https://github.com/AcalaNetwork/acala-node/releases instead for new node upgrades for acala and karura?

Yes. 

https://github.com/AcalaNetwork/Acala is for forkless runtime upgrade.
https://github.com/AcalaNetwork/acala-node is karura/acala node.

Now it is separated into two repositories.

**Reply by andreclaro** [reply]
All our nodes (Acala and Karura) were successfully upgraded. Thanks!
